ql/termstructures/yield/ratehelpers.hpp

```diff
@@ -369,8 +387,9 @@ namespace QuantLib {
                          const Calendar& calendar,
                          BusinessDayConvention convention,
                          bool endOfMonth,
-                         bool isFxBaseCurrencyCollateralCurrency,                   
-                         const Handle<YieldTermStructure>& collateralCurve);
+                         bool isFxBaseCurrencyCollateralCurrency,
+                         const Handle<YieldTermStructure>& collateralCurve,
+                         bool requireUSCalendar);
```

Comment:
Instead of passing a `bool` and hard-coding the class to use the US calendar, I'd pass the second calendar directly.  An empty `Calendar` instance can be passed when the second calendar is not required, and I'd set it as the default argument so that the interface remains backward-compatible.

Reply:
Regarding default argument, actually in  SWIG/ratehelpers.i I have implemented default value to false in order to provide this backward compatibility.
How about overloading the constructor? The reason why I decided for boolean indicator is that there exist only one calendar, that is used in this context. The first calendar which must be a JointCalendar of the two currencies involved is used for shifting the earliest date. Then, if there is no settlement in US, the three calendars including the US are used to shift the date. 
The only reason why alternative calendar would be allowed in this place would be a custom calendar, when QuantLib did not manage to include some new non-business day in UnitedStates::Settlement calendar.
Even if some additional business centers were to be added to date determination, I guess typically those would be related to the first calendar, not to pure handling of spot convention. There is no reason why some custom fx swap could be produced, but my understanding was that FxSwapRateHelper class is just a class used for replicating typical wholesale market fx swaps which are underpinning curve bootstrapping. The idea behind hard coding the specific calendar was just to take off from the user thinking about which of the US calendars would be the most appropriate. I must admit, that my selection was based on the name and did not perform thorough validation if this is the exact calendar which should be selected here.
Please let me know which approach would you prefer then. In case default boolean set to false is sufficient, it is already there.

Reply:
So, pardon my ignorance: this adjustment rule is only used with the US calendar? FX swaps quoted elsewhere don't use a similar rule with another calendar?